packit/rpkg.py

```diff
@@ -29,35 +30,52 @@
 from packit.utils.logging import logger
 
 
-class FedPKG:
+class PkgTool(Enum):
     """
-    Wrapper around fedpkg/centpkg which are wrappers around rpkg.
+    https://pagure.io/fedpkg
+    https://git.centos.org/centos/centpkg
+    """
+
+    fedpkg = "fedpkg"
+    # Staging dist-git & koji
+    fedpkg_stage = "fedpkg-stage"
+    # CentOS Stream
+    centpkg = "centpkg"
+    # CentOS Linux
+    centpkg_sig = "centpkg-sig"
```

Comment:
`rpkg` derivatives share the same CLI, approximately, so I'm wondering if this enum could be dropped entirely, have `fedpkg` as the name of default pkg-tool executable and just let the user to define whatever they want.

This would make `packit` more flexible and suport `rhpkg` (or any other tool named any other way or installed at any other path than the ones in `PATH`) without the need for a code change, as long as, that executable has the CLI interface the way we expect.

Reply:
Ok, I can drop that. I just thought it'd be nice to be able to verify that you don't have a typo in the name of the tool specified in the packit config by checking it's one of these.

Reply:
This is a valid concern, but if the executable is not found, there should be an error pointing to that. We definitely need to check this, so that we know how does it look like from a user POV.

Reply:
I don't understand what you're suggesting now. Can I keep it or should I drop it and do it some other way?

> error pointing to that.

to what?

> need to check this

check what?

Reply:
I'm saying you should drop it.

If someone makes a typo in the name of the tool, they will get `command not found...` error or similar when the `run_command` is called with the name. But I'm just guessing this, I haven't checked the code, so the actually behaviour might be different. This is why we should check what the actual behaviour is. By "check" I mean: to run the command and make it fail and see how the error looks like.

Reply:
Ok, removed.
It fails with
`ERROR  'wrongpkg new-sources' failed for the following reason: FileNotFoundError(2, 'No such file or directory')`

Reply:
Not bad.

Maybe it could be improved by checking with [`shutil.which`](https://docs.python.org/dev/library/shutil.html#shutil.which) if `wrongpkg` is a executable, and fail with a more explicit error (`wrongpkg is not an executable`, though @TomasTomecek might have a better idea). Do this as early as possible, ie. when type-checking the CLI option or the config value.

Reply:
done with `shutil.which`, thanks

Reply:
:partying_face: